backend/GymBeam/Controllers/UserController.cs

```diff
@@ -36,38 +35,44 @@ public async Task<ActionResult<IEnumerable<UserResponse>>> Get()
 #if !DEBUG
         [Authorize(Roles = Roles.Admin)]
 #endif
-        public ActionResult<UserResponse> Get(int id)
+        public async Task<ActionResult<UserResponse>> Get(int id)
         {
-            return new UserResponse
+            var request = new GetUserQuery
             {
-                Id = id,
-                Name = "testUsername",
-                DisplayName = "testDisplayName",
-                Role = "User"
+                UserId = id
             };
+            try
+            {
+                var result = await _mediator.Send(request);
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.BadRequest,
+                    $"BadRequest: {ex.Message}");
+            }
+
         }
 
         [HttpGet("LoggedIn")]
 #if !DEBUG
         [Authorize(Roles = Roles.User)]
 #endif
-        public ActionResult<UserResponse> GetLoggedIn()
+        public async Task<ActionResult<UserResponse>> GetLoggedIn()
         {
-            int userId;
+            int id;
             try
             {
                 string cookiesUserId = Request.Cookies[Cookies.UserId];
-                if (!int.TryParse(cookiesUserId, out userId))
+                if (!int.TryParse(cookiesUserId, out id))
                     throw new InvalidUserIdException();
 
-                return new UserResponse
+                var request = new GetUserQuery
```

Comment:
we can use code from get user by id from here. It is exactly the same as this one(after we get id of user from cookie)

Reply:
still not fixed